controller/internal/supervisor/supervisor.go

```diff
@@ -207,6 +209,17 @@ func (s *Config) doCreatePU(contextID string, pu *policy.PUInfo) error {
 	// Version the policy so that we can do hitless policy changes
 	s.versionTracker.AddOrUpdate(contextID, c)
 
+	var iprules policy.IPRuleList
+
+	iprules = append(iprules, pu.Policy.ApplicationACLs()...)
+	iprules = append(iprules, pu.Policy.NetworkACLs()...)
+
+	if err := ipsetmanager.RegisterExternalNets(contextID, iprules); err != nil {
+		s.Unlock()
+		zap.L().Error("Error creating ipsets for external networks", zap.Error(err))
+		return err
+	}
+
```

Comment:
Can't we do a RegisterPU method instead that takes the policy as argument and hides this inside it. No need for callers to know what to do for this.